### DISCUSSION TITLE
How to concatenate Strings in a loop

### DISCUSSION THREAD

**Discussion by subramanian0331** [cue-lang/cue#2182] [Q&A]
I want to concatenate string as i traverse through a loop.
`
cluster:{
    lab1:"val1"
    lab2:"val2"
    lab3: "val3"
} 

payload:{
    final : string
    for key,val in cluster{
        key + final
    }
}

`

**Comment by slewiskelly** [comment]
In CUE all concrete values are final, so you cannot concatenate. However, it seems like you can use [strings.Join](https://pkg.go.dev/cuelang.org/go/pkg/strings#Join):

```cue
import "strings"

cluster: {
	lab1: "val1"
	lab2: "val2"
	lab3: "val3"
}

payload: {
	final: strings.Join([ for k, _ in cluster {k}], "") // Results in "lab1lab2lab3".
}
```

When creating issues and/or discussion, I would suggest you clearly show what you expect/want to achieve, along with your examples.

Also, for future reference, [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) is a guide for formatting markdown in GitHub. This will be useful to better format issues/descriptions.
